scipy/stats/tests/test_distributions.py

```diff
@@ -3078,6 +3078,23 @@ def test_isf(self):
                1151387.578354072, 1640845512466.0906]
         assert_allclose(stats.loglaplace.isf(q, c), ref, rtol=1e-14)
 
+    @pytest.mark.parametrize('r, c',
+                             [(1, [0.5, 1.0]),
+                              (2, [0.5, 1.0, 1.5, 2.0]),
+                              (3, [0.5, 1.0, 1.5, 2.0, 2.5, 3.0]),
+                              (4, [0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0])])
+    def test_moment(self, r, c):
+        # r-th non-central moment is infinite if |r| >= c.
+        assert_almost_equal(stats.loglaplace.moment(r, c), np.inf)
+
+    @pytest.mark.parametrize('mom, c',
+                             [('m', [0.5, 1.0]),
+                              ('v', [0.5, 1.0, 1.5, 2.0]),
+                              ('s', [0.5, 1.0, 1.5, 2.0, 2.5, 3.0]),
+                              ('k', [0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0])])
+    def test_stats(self, mom, c):
+        # r-th non-central moment is non-finite (inf or nan) if r >= c.
+        assert not np.any(np.isfinite(stats.loglaplace.stats(c, moments=mom)))
```

Comment:
Would be a bit more concise to combine these parameterize over only the order of the moment `r`:
```python3
mom = 'mvsk'[r + 1]
c = np.linspace(0.5, r, 5)
```

Also, `assert_allclose` is preferred over [`assert_almost_equal`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html) (see doc).

Reply:
Good idea! Let me update accordingly.

Also I’d like to remark about the test for inf or nan. For central moment inf is well defined. For non-central moment I did a bit of googling and found there appears to lack a convention for choosing between nan and inf. For example, when mean is infinite, scipy’s implementation returns inf for var but nan for kurtosis. Since the distinction is not important for practical purpose, I simply test that the result is non-finite.

Reply:
I'll also combine these into one test.